src/plugins/intel_gpu/src/plugin/ops/constant.cpp

```diff
@@ -99,9 +99,14 @@ static void create_data(ProgramBuilder& p, const ov::Shape& const_shape, const s
         p.primitive_ids[initialconstPrimID] = constPrimID;
         p.profiling_ids.push_back(initialconstPrimID);
     } else {
+        bool allocate_mem = constLayout.count() <= cldnn::layout::max_rank();
```

Comment:
I don't understand the logic behind this comparison. How max rank is connected with elements count? It seems that you wanted to allow alloc for small constants, but you'd better compare it with some meaningful value

Reply:
As far as I noticed, small input constants mostly have a rank of 1, and a shape equal to or less than the rank of the input data. Could you suggest a value that will be more meaningful than the current one?

Reply:
It could be just a named constant hardcoded here. Also, I wonder if the different path for small constants is really needed? What are the benefits?